dags/materialized_columns.py

```diff
@@ -13,25 +13,28 @@
 from posthog import settings
 from posthog.clickhouse.cluster import AlterTableMutationRunner, ClickhouseCluster, HostInfo
 
-K = TypeVar("K")
+
+K1 = TypeVar("K1")
+K2 = TypeVar("K2")
 V = TypeVar("V")
 
 
-def zip_values(mapping: Mapping[K, Iterable[V]]) -> Iterator[Mapping[K, V]]:
-    """
-    Takes a mapping that contains values of sequences if identical lengths, returns an iterator of dictionaries for each
-    index in the sequence, with keys for each value taken from the input mapping.
-    """
-    keys, values = [], []
-    for key, value in mapping.items():
-        keys.append(key)
-        values.append(value)
+def join_mappings(mappings: Mapping[K1, Mapping[K2, V]]) -> Mapping[K2, Mapping[K1, V]]:
```

Comment:
This not a very good name for this function (it is actually a pretty bad one), but I'm struggling to come up with a good name that describes what this actually does. If you would like to help put me out of my misery, the type signature and tests are probably the best examples of what this is doing. Maybe this is something similar to transposition?